src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,127 @@
+                break;
+            case "done":
+                int numDone = Integer.parseInt(inputArr[1]);
+                Task task = tasks.list.get(numDone - 1);
+                tasks.checkAsDone(task);
+                ui.checkAsDoneMessage(task);
+                break;
+            case "delete":
+                int numDelete = Integer.parseInt(inputArr[1]);
+                Task deletedTask = tasks.deleteTask(numDelete);
+                ui.deleteTaskMessage(tasks.list, deletedTask);
+                break;
+            case "find":
+                String keyword = inputArr[1];
+                ui.findTask(keyword, tasks.list);
+                break;
+            case "todo":
+                try {
+                    parser.isEmptyDesc(inputArr);
+                    String details = inputArr[1];
+                    Task todo = new Todo(details);
+                    tasks.addTask(todo);
+                    ui.addTaskMessage(tasks.list, todo);
+                } catch (DukeException e) {
+                    System.out.println("OOPS!!! The description of a todo cannot be empty.\n" + HORIZONTAL_RULE);
+                }
+                break;
+            case "deadline":
+                try {
+                    parser.isEmptyDesc(inputArr);
+                    String details = input.substring(input.indexOf(" ") + 1, input.indexOf("/") - 1);
+                    String date = input.substring(input.indexOf("/") + 4);
+                    Task deadline = new Deadline(details, date);
+                    tasks.addTask(deadline);
+                    ui.addTaskMessage(tasks.list, deadline);
+                } catch (DukeException e) {
+                    System.out.println("OOPS!!! The description of a deadline cannot be empty.\n" + HORIZONTAL_RULE);
+                }
+                break;
+            case "event":
+                try {
+                    parser.isEmptyDesc(inputArr);
+                    String details = input.substring(input.indexOf(" ") + 1, input.indexOf("/") - 1);
+                    String date = input.substring(input.indexOf("/") + 4);
+                    Task event = new Event(details, date);
+                    tasks.addTask(event);
+                    ui.addTaskMessage(tasks.list, event);
+                } catch (DukeException e) {
+                    System.out.println("OOPS!!! The description of an event cannot be empty.\n" + HORIZONTAL_RULE);
+                }
+                break;
+            default:
+                try {
+                    throw new DukeException();
+                } catch (DukeException e) {
+                    System.out.println("OOPS!!! I'm sorry, but I don't know what that means :-(\n" + HORIZONTAL_RULE);
+                }
+            }
+        }
+        try {
+            storage.saveTasks(tasks.list);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        sc.close();
+    }
+
+}
```

Comment:
Overall code is very easy to read. It would be made easier to read if some method and variable names were longer and their short forms were not used, eg "numDone" and "isEmptyDesc"